pkg/artifactory/repositories.go

```diff
@@ -322,13 +329,30 @@ var repoTypesLikeGeneric = []string{
 	"vagrant",
 }
 
+var projectEnvironmentsSupported = []string{"DEV", "PROD"}
+
 var baseLocalRepoSchema = map[string]*schema.Schema{
 	"key": {
 		Type:         schema.TypeString,
 		Required:     true,
 		ForceNew:     true,
 		ValidateFunc: repoKeyValidator,
 	},
+	"project_key": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: projectKeyValidator,
+		Description:      "Project key for assigning this repository to. When assigning repository to a project, repository key must be prefixed with project key, separated by a dash.",
+	},
+	"project_environments": {
+		Type:        schema.TypeSet,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		MinItems:    1,
+		MaxItems:    2,
+		Set:         schema.HashString,
+		Optional:    true,
+		Description: `Project environment for assigning this repository to. Allow values: "DEV" or "PROD"`,
```

Comment:
Is this is the case, then you want to validate that, right? But my understanding is that project environments are entirely arbitrary